core/src/main/scala/org/apache/spark/sql/delta/commands/MergeIntoCommand.scala

```diff
@@ -548,26 +557,78 @@ case class MergeIntoCommand(
     }
 
     val joinedPlan = joinedDF.queryExecution.analyzed
+    val cdcEnabled = DeltaConfigs.CHANGE_DATA_FEED.fromMetaData(deltaTxn.metadata)
+
+    // ==== Generate the expressions to process full-outer join output and generate target rows ====
+    // If there are N columns in the target table, there will be N + 2 columns after processing
+    // - N columns for target table
+    // - INCR_ROW_COUNT_COL containing a UDF to update the output row row counter
+    // - CDC_TYPE_COLUMN_NAME containing the type of change being performed in a particular row
+
+    // To generate these N + 2 columns, we will generate N + 2 expressions and apply them to the
+    // rows in the joinedDF. The CDC column will be either used for CDC generation or dropped before
+    // performing the final write, and the increment column will always be dropped after executing
+    // the metrics UDF.
+
+    // We produce both rows for the CDC_TYPE_NOT_CDC partition to be written to the main table,
```

Comment:
itst hard to understand what partition means in the immediate context of this code. rewrite differently

Reply:
Updated. LMK if it's more clear.